File: backend/src/main/java/at/ac/htl/features/cpu/CPUMapper.java
package at.ac.htl.features.cpu;

import jakarta.enterprise.context.ApplicationScoped;

@ApplicationScoped
public class CPUMapper {
    public CPUDto toResource(CPU cpu) {
        return new CPUDto(
                cpu.cpu_id,
                cpu.name,
                cpu.price,
                cpu.core_count,
                cpu.core_clock,
                cpu.boost_clock,
                cpu.tdp,
                cpu.graphics,
                cpu.smt,
                cpu.socket,
                cpu.img,
                cpu.model
        );
    }
}

File: android_frontend/app/src/main/java/at/htl/leonding/android_frontend/ui/screens/components/mapper/CpuMapper.kt
package at.htl.leonding.android_frontend.ui.screens.components.mapper

// FILE: ui/screens/cpu/CpuMappers.kt

import at.htl.leonding.android_frontend.data.model.CpuDto
import at.htl.leonding.android_frontend.ui.screens.components.ComponentListItem

fun CpuDto.toListItem(): ComponentListItem {
    val subtitle = when {
        coreClock != null && boostClock != null -> "${coreClock} GHz • Boost ${boostClock} GHz"
        coreClock != null -> "${coreClock} GHz"
        boostClock != null -> "Boost ${boostClock} GHz"
        else -> ""
    }
    return ComponentListItem(
        id = cpuId,
        title = name,
        subtitle = subtitle,
        imageUrl = img
    )
}
